scripts/extensions/sams/src/utils/assets.ts

```diff
@@ -93,3 +93,16 @@ export function getMimetypeHumanReadable(mimetype?: string): string {
         return gettext('Other');
     }
 }
+
+export function verifyAssetBeforeLocking(asset: Partial<IAssetItem>, lock_action?: string): boolean {
+    const session_id = superdeskApi.session.getSessionId();
+    const user_id_verified = superdeskApi.session.getCurrentUser().then((user) => {
```

Comment:
Because `session.getCurrentUser` returns a promise, this promise will not be resolved by the time it is used in the if condition below.

Maybe for now we can have a synchronous function `session.getCurrentUserId()` which simply returns `session.identity._id` (as `session.identity` is actually set upon login)